File: java/sqlbuilder/GroupBy.java
package com.clickability.cms.dataaccess.sqlbuilder;

import java.util.StringJoiner;

public class GroupBy implements Expression {

	private StringJoiner groupBy;
	
	public GroupBy() {
		this.groupBy = new StringJoiner(",");
	}
	
	public GroupBy(Column column) {
		this();
		add(column);
	}
	
	public GroupBy(String column) {
		this();
		add(column);
	}
	
	public GroupBy add(Column column) {
		this.groupBy.add(column.get());
		return this;
	}
	
	public GroupBy add(String column) {
		this.groupBy.add(column);
		return this;
	}
	
	@Override
	public String get() {
		return String.format("group by %s", this.groupBy.toString());
	}


}

File: queryBuilder/src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/GroupBy.kt
package com.clickability.cms.dataaccess.sqlbuilder

import java.util.StringJoiner

class GroupBy() : Expression {

    private val groupBy: StringJoiner

    init {
        this.groupBy = StringJoiner(",")
    }

    constructor(column: Column) : this() {
        add(column)
    }

    constructor(column: String) : this() {
        add(column)
    }

    fun add(column: Column): GroupBy {
        this.groupBy.add(column.get())
        return this
    }

    fun add(column: String): GroupBy {
        this.groupBy.add(column)
        return this
    }

    @Override
    fun get(): String {
        return String.format("group by %s", this.groupBy.toString())
    }


}
